agent/dns.go

```diff
@@ -838,23 +863,33 @@ func (d *DNSServer) trimDomain(query string) string {
 	return strings.TrimSuffix(query, shorter)
 }
 
-// computeRCode Return the DNS Error code from Consul Error
-func (d *DNSServer) computeRCode(err error) int {
-	if err == nil {
+// rCodeFromError return the DNS Error code an error
+func rCodeFromError(err error) int {
+	switch {
+	case err == nil:
 		return dns.RcodeSuccess
-	}
-	if structs.IsErrNoDCPath(err) || structs.IsErrQueryNotFound(err) {
+	case errors.Is(err, errNoAnswer):
+		// TODO: why do we return success if the answer is empty?
+		return dns.RcodeSuccess
```

Comment:
I don't think we need to resolve this TODO, but I added it because I don't fully understand this case. The existing comments say:

> If the answer is empty and the response isn't truncated, return not found

However, we were never setting the response code to `RcodeNameError`. Maybe the existing comment is misleading? The one thing we did for this case was set the SOA response, and we still do that, so this preserves existing behaviour.

Does anyone know why this we set SOA for this case, but not the response code?